tardis/tests/fixtures/regression_data.py

```diff
@@ -161,9 +162,14 @@ def sync_hdf_store(self, tardis_module, update_fname=True):
                 f"Skipping test to generate regression data: {self.fpath}"
             )
         else:
-            return pd.HDFStore(self.fpath, mode="r")
-
+            # since each test function has its own regression data instance
+            # each test function will only have one HDFStore object
+            self.hdf_store_object = pd.HDFStore(self.fpath, mode="r")
+            return self.hdf_store_object
```

Comment:
Should we just create a dictionary here and update the tests accordingly?

Reply:
Since there is only one file for each test I don't think dict would do much help right?